### PR TITLE
HYRAX-667, change NULL to nullptr for the HDF4 handler. This can redu…

### PR DESCRIPTION
**Pull request by kyang2014**
…ce the code smell and also easy to find other issues from sonar cloud.

### CODE REVIEW

File: modules/hdf4_handler/HDF4RequestHandler.cc [hunk 1]
```diff
@@ -327,7 +327,7 @@ cerr<<"Don't output ecs metadata "<<endl;
                 //throw InternalErr(__FILE__,__LINE__,"read_das_use_eos2lib error");
 #endif
             }
-            if(eosfile != NULL)
+            if(eosfile != nullptr)
```
Comment (jgallagher59701) [inline modules/hdf4_handler/HDF4RequestHandler.cc:330]: There's no pressing need to change this, but as you write new code, there is no need to test if the pointers are not null before calling 'delete' since the delete operator can safely be used on null pointers (unlike free).

**Review by jgallagher59701 — APPROVED** [review]
As I said in my one comment, no need to change those - the gain in efficiency is very small, but in the future you can just dall 'delete pointer' and drop the test since delete is safe to use with null pointers.

File: modules/hdf4_handler/HDF4RequestHandler.cc (hunk 1 above)
Comment (kyang2014) [reply modules/hdf4_handler/HDF4RequestHandler.cc:330]: Thanks,  this will be handled later and I am going to remove the "new delete" with unique_pointer later. There are other cases that NULL needs to be changed to nullptr (like passing NULL as a parameter) if we want to conform to C11.  Also sonor cloud gives a lot of code smell. It buries other good warnings.

> here is no need to test if the pointers are not null before calling 'delete' since the delete operator can safely be used on null pointers (unlike free).
